### ISSUE TITLE
Group `End` and `Transition` properties

### ISSUE THREAD

**Issue by cdavernas** [serverlessworkflow/specification#684]
**What would you like to be added**:
Group `End` and `Transition` properties, in such way that we can address it as an (activity | state | workflow) outcome.

In other words, what I'd like to achieve is:

```yaml
...
states:
  - name: MyState
    outcome:
      type: end
      produceEvents: ...
    ...
...
```

**Why is this needed**:

Because `End` and `Transition` properties are mutually exclusive, and implicitly defines the outcome of an (activity | state | workflow), it makes sense to call is as such. In addition, it simplifies processing of the whole definition, by checking that an outcome has been specified instead of having to do checks on multiple properties. @tsurdilo WDYT?

**Comment by tsurdilo** [comment]
Yes, i wish we did this to begin with. Will update

**Comment by cdavernas** [comment]
Closed as resolved by 0.10, and therefore as part of #843
